botocore/regions.py

```diff
@@ -433,3 +448,294 @@ class EndpointResolverBuiltins(str, Enum):
+
+            signing_context = {}
+            if 'signingRegion' in scheme:
+                signing_context['region'] = scheme['signingRegion']
+            elif 'signingRegionSet' in scheme:
+                # todo: can we handle lists of regions?
+                if len(scheme['signingRegionSet'][0]) > 0:
+                    signing_context['region'] = scheme['signingRegionSet'][0]
+            if 'signingName' in scheme:
+                signing_context.update(signing_name=scheme['signingName'])
+            if 'disableDoubleEncoding' in scheme:
+                signing_context['disableDoubleEncoding'] = ensure_boolean(
+                    scheme['disableDoubleEncoding']
+                )
+
+            return scheme['name'], signing_context
+        else:
+            # If an authSchemes list is present in the Endpoint object but none
+            # of the entries are supported, raise an exception. (However, if
+            # authSchemes is not present, use the service's default auth type.)
+            fixable_with_awscrt = False
+            auth_type_options = [scheme['name'] for scheme in auth_schemes]
+            if not HAS_CRT:
+                from botocore.crt.auth import CRT_AUTH_TYPE_MAPS
+
+                for auth_type in auth_type_options:
+                    if auth_type in CRT_AUTH_TYPE_MAPS:
+                        fixable_with_awscrt = True
+
+            if fixable_with_awscrt:
+                raise MissingDependencyException(
+                    msg='This operation requires an additional dependency. '
+                    'Use pip install botocore[crt] before proceeding.'
+                )
+            else:
+                raise UnknownSignatureVersionError(
+                    signature_version=', '.join(auth_type_options)
+                )
+
+    def ruleset_error_to_botocore_exception(self, ruleset_exception, params):
+        """Attempts to translate ruleset errors to pre-existing botocore
+        exception types by string matching exception strings.
+        """
+        msg = ruleset_exception.kwargs.get('msg')
+        if msg is None:
+            return
+        service_name = self._service_model.service_name
+
+        if service_name == 's3':
+            if (
+                msg.startswith('S3 Outposts does not support')
+                or msg.startswith('S3 MRAP does not support')
+                or msg.startswith('S3 Object Lambda does not support')
+                or msg.startswith('Access Points do not support')
+                or msg.startswith('Invalid configuration:')
+                or msg.startswith('Client was configured for partition')
+            ):
+                return UnsupportedS3AccesspointConfigurationError(msg=msg)
+            if msg == 'S3 Object Lambda does not support S3 Accelerate':
+                return UnsupportedS3ConfigurationError(msg=msg)
+            if msg.startswith('Invalid ARN:') or msg.startswith(
+                'Invalid Arn:'
+            ):
```

Comment:
Are we trying to avoid other casing variants being captured here, or could we do something like:
```suggestion
            if msg.lower().startswith('invalid arn:'):
```

Also this seems like a potential issue with the error messages themselves? They should be consistent with `ARN` not `Arn`.